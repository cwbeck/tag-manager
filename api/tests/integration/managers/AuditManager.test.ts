import ManagerTestUtils from '../../utils/ManagerTestUtils';
import TestUtils from '../../utils/TestUtils';
import AuditRepo from '../../../src/mongo/repos/AuditRepo';
import Mock = jest.Mock;

describe('AuditManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('gets HistoryForEntities', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockOrg = TestUtils.buildMockOrg();
        const mockAudit1 = TestUtils.buildMockAudit(mockOrg);
        const mockAudit2 = TestUtils.buildMockAudit(mockOrg, 'aaaf4c48e9615d9a1c66bccc');
        const mockSutRepo = {} as AuditRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockAudit1);
        mockSutRepo.find = jest.fn().mockResolvedValue([mockAudit1, mockAudit2]);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getHistoryForEntities(entities: ["5edf4c48e9615d9a1c66bc2a"]) {
                    id
                    user {
                        id
                        first_name
                        last_name
                    }
                    owner
                    model_id
                    model
                    model_persisting_id
                    connected_models {
                        id
                        type
                    }
                    action
                    method
                    comments
                    created_at
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
