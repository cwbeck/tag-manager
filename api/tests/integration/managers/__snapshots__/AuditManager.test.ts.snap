// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditManager gets HistoryForEntities 1`] = `
Object {
  "data": Object {
    "getHistoryForEntities": Array [
      Object {
        "action": "Create",
        "comments": "comment",
        "connected_models": Array [
          Object {
            "id": "5edf4c48e9615d9a1c66bbbb",
            "type": "Org",
          },
        ],
        "created_at": 1608175440000,
        "id": "5edf4c48e9615d9a1c66bccc",
        "method": "INDIRECT_METHOD",
        "model": "Org",
        "model_id": "5edf4c48e9615d9a1c66bbbb",
        "model_persisting_id": "pId",
        "owner": "SYSTEM",
        "user": null,
      },
      Object {
        "action": "Create",
        "comments": "comment",
        "connected_models": Array [
          Object {
            "id": "5edf4c48e9615d9a1c66bbbb",
            "type": "Org",
          },
        ],
        "created_at": 1608175440000,
        "id": "aaaf4c48e9615d9a1c66bccc",
        "method": "INDIRECT_METHOD",
        "model": "Org",
        "model_id": "5edf4c48e9615d9a1c66bbbb",
        "model_persisting_id": "pId",
        "owner": "SYSTEM",
        "user": null,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
